platforms/core-runtime/logging/src/main/java/org/gradle/internal/deprecation/DeprecationMessageBuilder.java

```diff
@@ -390,13 +393,17 @@ String formatAdvice(List<String> replacements) {
         }
     }
 
+    static final Pattern ID_CLEAN_UP_PATTERN = Pattern.compile("[^a-z-]+");
+
     public static String createDefaultDeprecationId(String... ids) {
         // the replaceAll calls are to ensure that the id is a valid identifier.
         // it removes all whitespace and non-alphanumeric characters, and replaces multiple dashes with a single dash.
```

Comment:
❌ The comment should be fixed.

>  non-alphanumeric characters

It also removes numbers as shown by the test

> replaces multiple dashes

but dashes are non-alphanumeric